clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureResourceManagerClient.groovy

```diff
@@ -62,12 +62,13 @@ class AzureResourceManagerClient extends AzureBaseClient {
                                                 String resourceGroupName,
                                                 String region,
                                                 String resourceName,
+                                                String resourceType,
                                                 Map<String, String> templateParams = [:]) {
 
     // TODO validate that all callers invoke this themselves, then remove this call
     initializeResourceGroupAndVNet(credentials, resourceGroupName, null, region)
 
-    String deploymentName = resourceName + AzureUtilities.NAME_SEPARATOR +"deployment"
+    String deploymentName = resourceName + AzureUtilities.NAME_SEPARATOR + resourceType + AzureUtilities.NAME_SEPARATOR +"deployment"
```

Comment:
Not a big deal but `[resourceName, resourceType, "deployment"].join(AzureUtilities.NAME_SEPARATOR]` would be another way to write this.